### ISSUE TITLE
cannot be shared between threads safely

### ISSUE THREAD

**Issue by aep** [seanmonstar/futures-fs#17]
how do you actually use this with tokio which requires all futures to impl Send?

```
293 |             tokio::run(futures::lazy(move || {
    |             ^^^^^^^^^^ `(dyn futures::future::Executor<std::boxed::Box<(dyn futures::Future<Error=(), Item=()> + std::marker::Send + 'static)>> + 'static)` cannot be shared between threads safely
    |
    = help: the trait `std::marker::Sync` is not implemented for `(dyn futures::future::Executor<std::boxed::Box<(dyn futures::Future<Error=(), Item=()> + std::marker::Send + 'static)>> + 'static)`
    = note: required because of the requirements on the impl of `std::marker::Send` for `std::sync::Arc<(dyn futures::future::Executor<std::boxed::Box<(dyn futures::Future<Error=(), Item=()> + std::marker::Send + 'static)>> + 'static)>`
    = note: required because it appears within the type `futures_fs::FsPool`
    = note: required because it appears within the type `futures_fs::FsReadStream`
    = note: required because it appears within the type `[closure@cli/src/main.rs:452:55: 477:6 target:carrier::Identity, secret:carrier::Secret, remote_file:std::string::String, read:futures_fs::FsReadStream]`
    = note: required because it appears within the type `futures::future::chain::Chain<impl futures::Future, futures::AndThen<impl futures::Future, futures::AndThen<futures::MapErr<futures::Flatten<futures::AndThen<futures::MapErr<futures::stream::StreamFuture<carrier::channel::ChannelStream>, [closure@cli/src/main.rs:462:26: 462:36]>, std::result::Result<futures::stream::Forward<futures::stream::MapErr<futures_fs::FsReadStream, fn(std::io::Error) -> failure::Error {<failure::Error as std::convert::From<std::io::Error>>::from}>, carrier::channel::ChannelStream>, failure::Error>, [closure@cli/src/main.rs:463:27: 468:18 read:futures_fs::FsReadStream]>>, fn(failure::Error) -> failure::Error {<failure::Error as std::convert::From<failure::Error>>::from}>, std::result::Result<(), failure::Error>, [closure@cli/src/main.rs:471:27: 475:18 brk:carrier::channel::Channel, channel:carrier::channel::Channel]>, [closure@cli/src/main.rs:454:80: 476:10 remote_file:std::string::String, read:futures_fs::FsReadStream, brk:carrier::channel::Channel]>, [closure@cli/src/main.rs:452:55: 477:6 target:carrier::Identity, secret:carrier::Secret, remote_file:std::string::String, read:futures_fs::FsReadStream]>`
    = note: required because it appears within the type `futures::AndThen<impl futures::Future, futures::AndThen<impl futures::Future, futures::AndThen<futures::MapErr<futures::Flatten<futures::AndThen<futures::MapErr<futures::stream::StreamFuture<carrier::channel::ChannelStream>, [closure@cli/src/main.rs:462:26: 462:36]>, std::result::Result<futures::stream::Forward<futures::stream::MapErr<futures_fs::FsReadStream, fn(std::io::Error) -> failure::Error {<failure::Error as std::convert::From<std::io::Error>>::from}>, carrier::channel::ChannelStream>, failure::Error>, [closure@cli/src/main.rs:463:27: 468:18 read:futures_fs::FsReadStream]>>, fn(failure::Error) -> failure::Error {<failure::Error as std::convert::From<failure::Error>>::from}>, std::result::Result<(), failure::Error>, [closure@cli/src/main.rs:471:27: 475:18 brk:carrier::channel::Channel, channel:carrier::channel::Channel]>, [closure@cli/src/main.rs:454:80: 476:10 remote_file:std::string::String, read:futures_fs::FsReadStream, brk:carrier::channel::Channel]>, [closure@cli/src/main.rs:452:55: 477:6 target:carrier::Identity, secret:carrier::Secret, remote_file:std::string::String, read:futures_fs::FsReadStream]>`
    = note: required because it appears within the type `impl futures::Future`
    = note: required because it appears within the type `futures::MapErr<impl futures::Future, [closure@cli/src/main.rs:294:81: 294:100]>`
    = note: required because it appears within the type `futures::future::lazy::_Lazy<[closure@cli/src/main.rs:293:38: 295:14 secrets:_, target:_, local_file:_, remote_file:_], futures::MapErr<impl futures::Future, [closure@cli/src/main.rs:294:81: 294:100]>>`
    = note: required because it appears within the type `futures::Lazy<[closure@cli/src/main.rs:293:38: 295:14 secrets:_, target:_, local_file:_, remote_file:_], futures::MapErr<impl futures::Future, [closure@cli/src/main.rs:294:81: 294:100]>>`
    = note: required by `tokio::run`

```

**Comment by seanmonstar** [comment]
Well then, it looks like the 0.0.4 release mistakenly doesn't require the internal executor to be `Send + Sync`, so you see this issue. I'll make a new release fixing that.

**Comment by seanmonstar** [comment]
Published a 0.0.5 that is now `Send + Sync`.

**Comment by aep** [comment]
thanks!
